src/source/src/manager.rs

```diff
@@ -111,12 +113,12 @@ pub struct SourceDesc {
     pub pk_column_ids: Vec<i32>,
 }
 
-pub type SourceManagerRef = Arc<dyn SourceManager>;
+pub type TableSourceManagerRef = Arc<dyn TableSourceManager>;
 
 #[derive(Debug, Default)]
 struct MemSourceManagerInner {
     sources: HashMap<TableId, SourceDesc>,
-    source_actor_num: HashMap<TableId, usize>,
+    source_ref_count: HashMap<TableId, WeakSourceRef>,
```

Comment:
How about directly storing `HashMap<TableId, Weak<SourceDesc>>` and returning `Arc<SourceDesc>`?

Reply:
Sounds nice. Done it.